config/peerpods/podvm/packer-resource-cleanup.sh

```diff
@@ -98,12 +97,53 @@ function azure_cleanup() {
     done
 }
 
+function aws_cleanup() {
+    log "Deleting Packer VMs"
+
+    # Get the default VPC for the region. This is created as part of packer build.
+    DEFAULT_VPC_ID=$(aws ec2 describe-vpcs --filters "Name=is-default,Values=true" --region "$AWS_REGION" --query "Vpcs[0].VpcId" --output text)
+    log "Default VPC: $DEFAULT_VPC_ID"
+
+    aws ec2 describe-instances --filters "Name=instance.group-name,Values=packer*" \
+        "Name=vpc-id,Values=$DEFAULT_VPC_ID" \
+        --region "$AWS_REGION" \
+        --query "Reservations[*].Instances[*].[InstanceId, KeyName, join(',', SecurityGroups[*].GroupId)]" \
+        --output text | while read -r INSTANCE_ID KEY_NAME SG_IDS; do
+
+        log "Deleting"
+        log "Instance ID: $INSTANCE_ID"
+        log "Key Name: $KEY_NAME"
+        log "Security Groups: $SG_IDS"
+
+        # Terminate the instance
+        aws ec2 terminate-instances --instance-ids "$INSTANCE_ID" --region "$AWS_REGION"
+
+        # Wait for instance to be completely terminated
+        echo "Waiting for instance $INSTANCE_ID to terminate..."
+        aws ec2 wait instance-terminated --instance-ids "$INSTANCE_ID" --region "$AWS_REGION"
+
+        # Delete the key pair if it exists
+        if [[ -n "$KEY_NAME" && "$KEY_NAME" != "None" ]]; then
+            aws ec2 delete-key-pair --key-name "$KEY_NAME" --region "$AWS_REGION"
+        fi
+
+        # Delete the security groups (iterate if multiple SGs exist)
+        for SG_ID in $(echo "$SG_IDS" | tr ',' ' '); do
+            aws ec2 delete-security-group --group-id "$SG_ID" --region "$AWS_REGION"
+        done
```

Comment:
Side questions given other providers are not doing that either, but IIUC this cleanup might actually interrupt podvm creation, which might lead to partial resources being created (like image uploaded to s3 bucket). Do we care about these as well since we are the ones interrupting the VMs? (I haven't seen anything like that for AWS case so I guess we don't care at this point?).

Reply:
I don't know the possible workflows, but what I had in mind is:

1. podvm creation starts
2. before it's done the user start uninstall
3. this script starts killing the VMs (and now it occurred to me that the VM is not the builder pod so we should be fine)

So yes, it should be fine killing the VM provided the builder pod takes care of the build failure (and as it should not have any image ready yet it should be fine).